crates/swc_plugin_runner/src/transform_executor.rs

```diff
@@ -102,14 +102,26 @@ impl TransformExecutor {
         }
     }
 
+    /**
+     * Check compile-time versions of AST schema between the plugin and
+     * the host. Returns true if it's compatible, false otherwise.
+     *
+     * Host should appropriately handle if plugin is not compatible to the
+     * current runtime.
+     */
+    pub fn is_transform_schema_compatible(&self) -> bool {
+        todo!("Not supported yet");
+    }
+
     #[tracing::instrument(level = "info", skip_all)]
     pub fn transform(
         &mut self,
```

Comment:
Can we take `self` instead?
As it's single-run only, it's better to take ownership.

Reply:
I'd like to see with further refactorings - I am under assumption I may need to touch few other places until implementing version schema. Once it's done I can determine if this is one-off and can take self completely.